src/dxtbx/nexus/__init__.py

```diff
@@ -273,19 +275,33 @@ def get_dxtbx_detector(
                 reversed_dependency_chain = reversed(
                     nxmx.get_dependency_chain(module.fast_pixel_direction.depends_on)
                 )
-                pg: dxtbx.model.Detector | dxtbx.model.Panel = root
-                for transformation in reversed_dependency_chain:
-                    assert isinstance(
-                        pg, (dxtbx.model.Detector, dxtbx.model.DetectorNode)
-                    )
-                    name = transformation.path
-                    pg_names = [child.get_name() for child in pg]
-                    if name in pg_names:
-                        pg = pg[pg_names.index(name)]  # Getitem always returns panel
-                        continue
+                pg: dxtbx.model.Detector | dxtbx.model.Panel | None = None
+
+                # Group any transformations together that share the same equipement_component
+                # to reduce the number of hierarchy levels
+                def equipment_component_key(dependency):
+                    return dependency.equipment_component
+
+                for _, transformation_group in groupby(
+                    reversed_dependency_chain, key=equipment_component_key
```

Comment:
@ndevenish commented offline that 

```
key=lambda dep: dep.equipment_component_key
```

would remove the need for the explicit function (which would also be in keeping with the functional nature of itertools)